servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -228,6 +235,22 @@ public Reference getReferenceByName(String refName, FetchOption fetchOption)
   public Reference createReference(
       String refName, Reference.ReferenceType type, String targetHash, String sourceRefName)
       throws NessieNotFoundException, NessieConflictException {
+    BatchAccessChecker check =
+        startAccessCheck().canCreateReference(RefUtil.toNamedRef(type, refName));
+    try {
+      check.canViewReference(namedRefWithHashOrThrow(sourceRefName, targetHash).getValue());
+    } catch (NessieNotFoundException e) {
+      // If the default-branch does not exist and hashOnRef points to the "beginning of time",
+      // then do not throw a NessieNotFoundException, but re-create the default branch. In all
+      // cases, re-throw the exception.
+      if (!(Reference.ReferenceType.BRANCH.equals(type)
+          && refName.equals(getConfig().getDefaultBranch())
```

Comment:
So we do not run the default branch re-creation requests through access checks? Is that intended?

Would it be preferable to call `namedRefWithHashOrThrow` (line 241) separately and always pass the branch name to `check.canViewReference()`?

Reply:
The "hard check" is intentional.